tests/test_edge_cases_comprehensive.py

```
@@ -0,0 +1,454 @@
+            mock_anthropic.return_value = mock_client
+
+            claude_client = ClaudeClient()
+
+            # Should handle timeout gracefully
+            with self.assertRaises(Exception):
+                claude_client.analyze_text("Test text", "Test prompt")
+
+    def test_api_rate_limiting_simulation(self):
+        """Test handling of API rate limiting."""
+        # Mock rate limiting scenario
+        with patch('time.sleep') as mock_sleep:
```

Ruff (0.12.2)

Local variable `mock_sleep` is assigned to but never used

Remove assignment to unused variable `mock_sleep`

(F841)